src/App.js

```diff
@@ -14,11 +16,107 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const INITIAL_CRITERIA = {
+  isStartScript: true,
+  isReversed: false,
+  sortBy: null,
+};
+
+class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
+    ...INITIAL_CRITERIA,
+  };
+
+  startSript = () => this.setState({ isStartScript: false });
+
+  sortAlphabetically = () => this.setState({ sortBy: 'alphabetically' });
+
+  sortByLength = () => this.setState({ sortBy: 'byLength' });
+
+  reset = () => {
+    this.setState(state => ({
+      ...state,
+      ...INITIAL_CRITERIA,
+      isStartScript: false,
+    }));
+  }
+
+  reverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  }
+
+  render() {
+    const { isStartScript, goods, isReversed, sortBy } = this.state;
+    const goodsByCriteria = [...goods];
+
+    goodsByCriteria.sort((a, b) => {
+      switch (sortBy) {
+        case 'byLength':
+          return a.length - b.length;
+        case 'alphabetically':
+          return a.localeCompare(b);
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      goodsByCriteria.reverse();
+    }
+
+    return (
+      <div className="App">
+        <button
```

Comment:
```suggestion
        {isStartScript && <button
```
Better to use conditional rendering instead of setting `display: none`